src/AnalysisManager.cpp

```diff
@@ -27,320 +48,159 @@ AnalysisManager::~AnalysisManager()
-    generator_final_particle_charge_.push_back(particle->Charge());
-    generator_final_particle_x_.push_back(particle->X());
-    generator_final_particle_y_.push_back(particle->Y());
-    generator_final_particle_z_.push_back(particle->Z());
-    generator_final_particle_t_.push_back(particle->T());
-    generator_final_particle_px_.push_back(particle->Px());
-    generator_final_particle_py_.push_back(particle->Py());
-    generator_final_particle_pz_.push_back(particle->Pz());
-    generator_final_particle_energy_.push_back(particle->Energy());
-}
-
-//-----------------------------------------------------------------------------
-void AnalysisManager::AddMCParticle(MCParticle const * particle)
-{
-    particle_track_id_.push_back(particle->TrackID());
-    particle_parent_track_id_.push_back(particle->ParentTrackID());
-    particle_pdg_code_.push_back(particle->PDGCode());
-    particle_mass_.push_back(particle->Mass());
-    particle_charge_.push_back(particle->Charge());
-    particle_process_key_.push_back(this->ProcessToKey(particle->Process()));
-    particle_total_occupancy_.push_back(particle->TotalOccupancy());
-
-    particle_initial_x_.push_back(particle->InitialPosition().X());
-    particle_initial_y_.push_back(particle->InitialPosition().Y());
-    particle_initial_z_.push_back(particle->InitialPosition().Z());
-    particle_initial_t_.push_back(particle->InitialPosition().T());
-
-    particle_initial_px_.push_back(particle->InitialMomentum().X());
-    particle_initial_py_.push_back(particle->InitialMomentum().Y());
-    particle_initial_pz_.push_back(particle->InitialMomentum().Z());
-    particle_initial_energy_.push_back(particle->InitialMomentum().E());
-
-    particle_number_daughters_.push_back(particle->NumberDaughters());
-    particle_daughter_track_ids_.push_back(particle->Daughters());
-
-    number_particles_ += 1;
-
-    std::vector< TrajectoryHit > const hits = particle->Hits();
-
-    for (auto const & hit : hits)
-    {
-        energy_deposit_ += hit.Energy();
-
-        hit_track_id_.push_back(hit.TrackID());
-
-        hit_start_x_.push_back(hit.StartPoint().X());
-        hit_start_y_.push_back(hit.StartPoint().Y());
-        hit_start_z_.push_back(hit.StartPoint().Z());
-        hit_start_t_.push_back(hit.StartTime());
-
-        hit_end_x_.push_back(hit.EndPoint().X());
-        hit_end_y_.push_back(hit.EndPoint().Y());
-        hit_end_z_.push_back(hit.EndPoint().Z());
-        hit_end_t_.push_back(hit.EndTime());
-
-        hit_length_.push_back(hit.Length());
-        hit_energy_deposit_.push_back(hit.Energy());
-
-        hit_process_key_.push_back(this->ProcessToKey(hit.Process()));
-        number_hits_ += 1;
-    }
+  G4AutoLock metaLock(&metaMutex);
+  detector_length_x_ = detector_length_x;
+  detector_length_y_ = detector_length_y;
+  detector_length_z_ = detector_length_z;
+  detectorConfiguration_ = ConfigManager::GetDetectorConfiguration();
```

Comment:
This method is not implemented in the ConfigManager class.

Reply:
looks like this is resolved with `ConfigManager::Instancer();`
```suggestion
  detectorConfiguration_ = ConfigManager::Instance();
```

Reply:
There should be more in ConfigManager.h. Let me see whats going on with this

Reply:
These changes were reverted and fixed properly by adding some uncommitted changes from Dave's clone.